### PR TITLE
Add type annotation to contextItemExpr

### PR DESCRIPTION
**Pull request by FBroy**
# Description

Add type annotation to the contextItemExpr.

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/typeInference/annotateContextItemExpr.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { SequenceType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+/**
+ * A context item expression evaluates to the context item. Hence the type that is returned is the one from the context item.
+ * @param ast the AST to be annotated.
+ * @returns the type of the context item.
+ */
+export function annotateContextItemExpr(ast: IAST): SequenceType | undefined {
+	const typeNode = astHelper.followPath(ast, ['type']);
```
Comment (maanpalle) [inline src/typeInference/annotateContextItemExpr.ts]: This will always return undefined as the type is not annotated

**Review by HectorPeeters — APPROVED** [review]
If you just return undefined this looks good!

**Review by yuanzexiong — APPROVED** [review]
Let's just open an issue on this one so we won't forget to actually return the relevant values instead of an undefined

File: src/typeInference/annotateContextItemExpr.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { SequenceType } from '../expressions/dataTypes/Value';
+import { IAST } from '../parsing/astHelper';
+
+/**
+ * A context item expression evaluates to the context item. Hence the type that is returned is the one from the context item.
+ * @param ast the AST to be annotated.
+ * @returns the type of the context item.
+ */
+export function annotateContextItemExpr(ast: IAST): SequenceType | undefined {
```
Comment (MTimmerije) [inline src/typeInference/annotateContextItemExpr.ts:9]: This might warrant a new issue, but alas

**Review by MTimmerije — APPROVED** [review]

File: src/typeInference/annotateContextItemExpr.ts (hunk 2 above)
Comment (MTimmerije) [reply src/typeInference/annotateContextItemExpr.ts:9]: Apparently it is in todos already, so scratch that.

**Pull request merged**